tests/test_query.py

```diff
@@ -406,14 +386,156 @@ def test_check_user_resource_paths_status(client, list_policies_patcher, test_da
 
     # check whether there is a match
     # (True if the access is re-requestable, False otherwise)
+    data = {"resource_paths": [test_data["resource_path"]]}
+    res = client.post(
+        "/request/user_resource_paths",
+        json=data,
+        headers={"Authorization": f"bearer {fake_jwt}"},
+    )
+    assert res.status_code == 200, res.text
+    assert res.json() == {test_data["resource_path"]: test_data["should_match"]}
+
+
+@pytest.mark.parametrize(
+    "test_data",
+    [
+        {
+            "policy_id": "test-policy",
+            "resource_path": "/a",
+            "permissions": [
+                {
+                    "id": "original-permissions-1",
+                    "description": "",
+                    "action": {
+                        "service": "*",
+                        "method": "write",
+                    },
+                },
+                {
+                    "id": "original-permissions-2",
+                    "description": "",
+                    "action": {
+                        "service": "*",
+                        "method": "delete",
+                    },
+                },
+            ],
+        },
+    ],
+)
+def test_check_permissions_mismatch(client, list_policies_patcher, test_data):
+    fake_jwt = "1.2.3"
+
+    # create a request with an active status
+    data = {
+        "policy_id": test_data["policy_id"],
+        "resource_id": "uniqid",
+        "resource_display_name": "My Resource",
+        "status": "APPROVED",
+    }
+    res = client.post(
+        "/request", json=data, headers={"Authorization": f"bearer {fake_jwt}"}
+    )
+    assert res.status_code == 201, res.text
+
+    # * Permissions provided and matching
     data = {
         "resource_paths": [test_data["resource_path"]],
-        "permissions": test_data["permissions"],
+        "permissions": [
+            {
+                "id": "original-permissions-1",
+                "description": "",
+                "action": {
+                    "service": "*",
+                    "method": "write",
+                },
+            },
+            {
+                "id": "original-permissions-2",
+                "description": "",
+                "action": {
+                    "service": "*",
+                    "method": "delete",
+                },
+            },
+        ],
```

Comment:
i think there was a misunderstanding, the endpoint still takes permission IDs as input, not the whole permission data